MonoGame.Framework/Media/MediaPlayer.WMS.cs

```diff
@@ -124,29 +137,39 @@ private static void PlatformPlaySong(Song song)
             // Cleanup the last song first.
             if (State != MediaState.Stopped)
             {
-				_session.Stop();
+                _session.Stop();
                 _session.ClearTopologies();
                 _session.Close();
                 _volumeController.Dispose();
                 _clock.Dispose();
-			}
+            }
 
             // Set the new song.
             _session.SetTopology(SessionSetTopologyFlags.Immediate, song.Topology);
 
-            _volumeController = CppObject.FromPointer<SimpleAudioVolume>(VideoPlayer.GetVolumeObj(_session));
-            _volumeController.Mute = _isMuted;
-            _volumeController.MasterVolume = _volume;
+            // Get the volume interface.
+            IntPtr volumeObj;
+            try
+            {
+                MediaFactory.GetService(_session, MRPolicyVolumeService, AudioStreamVolumeGuid, out volumeObj);
+            }
+            catch
+            {
+                MediaFactory.GetService(_session, MRPolicyVolumeService, AudioStreamVolumeGuid, out volumeObj);
```

Comment:
So the code we already have that is similar to this yet improved:

https://github.com/mono/MonoGame/blob/develop/MonoGame.Framework/Media/VideoPlayer.WMS.cs#L126

Just a different GUID, but same sort of issue.

@KonajuGames - Should we maybe call this from here too?